code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -17,54 +15,93 @@
+	return
 
-					qdel(src)
-				return
+/obj/item/weapon/storage/lockbox/proc/toggle(var/mob/user, var/id_name)
+	if(allowed(user))
+		. = TRUE
+		locked = !locked
+		user.visible_message("<span class='notice'>The lockbox has been [locked ? null : "un"]locked by [user].</span>", "<span class='rose'>You [locked ? null : "un"]lock the box.</span>")
+		tracked_access = "The tracker reads: 'Last locked by [id_name || get_id_name(user)].'"
+		if(locked)
+			icon_state = icon_locked
 		else
-			to_chat(user, "<span class='warning'>Access Denied.</span>")
-	else if(istype(W, /obj/item/weapon/card/emag) && !src.broken)
-		broken = 1
-		locked = 0
-		desc = "It appears to be broken."
-		icon_state = src.icon_broken
-		for(var/mob/O in viewers(user, 3))
-			O.show_message(text("<span class='notice'>The lockbox has been broken by [] with an electromagnetic card!</span>", user), 1, text("You hear a faint electrical spark."), 2)
-		if(oneuse)
-			for(var/atom/movable/A in src)
-				remove_from_storage(A, get_turf(src))
+			icon_state = icon_closed
+	else
+		to_chat(user, "<span class='notice'>Access Denied.</span>")
+		return FALSE
+
+/proc/get_id_name(var/mob/user)
+	if (ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/O = H.get_item_by_slot(slot_wear_id)
+		var/obj/item/weapon/card/id/I = null
+		if (isPDA(O))
+			var/obj/item/device/pda/P = O
+			I = P.id
+		if (isID(O))
+			I = O
+		if (!I)
+			I = locate() in H.held_items
+		if (I)
+			return I.registered_name
+		else
+			return "UNKNOWN" // Shouldn't happen but eh
+	
+	else if (issilicon(user)) // Currently, borgos cannot open lockboxes, but if you want to make a module who can, this will work.
+		return "[user]"
 
-			qdel(src)
 
+/obj/item/weapon/storage/lockbox/oneuse/toggle(var/mob/user, var/id_name)
+	. = ..()
+	if (.)
+		for(var/atom/movable/A in src)
+			remove_from_storage(A, get_turf(src))
+		qdel(src)
+
+/obj/item/weapon/storage/lockbox/attackby(obj/item/weapon/W, mob/user)
+	if (isID(W))
+		var/obj/item/weapon/card/id/I = W
+		if(broken)
+			to_chat(user, "<span class='rose'>It appears to be broken.</span>")
+			return
+		return toggle(user, I.registered_name)
+	if (isPDA(W))
+		var/obj/item/device/pda/P = W 
+		var/obj/item/weapon/card/id/I = P.id
+		if (!I)
+			return
+		if(broken)
+			to_chat(user, "<span class='rose'>It appears to be broken.</span>")
+			return
+		return toggle(user, I.registered_name)
 	if(!locked)
 		. = ..()
 	else
 		to_chat(user, "<span class='warning'>It's locked!</span>")
-	return
+
+/obj/item/weapon/storage/lockbox/emag_act(var/mob/user)
+	if (broken)
+		return FALSE
+	broken = 1
+	locked = 0
+	desc = "It appears to be broken."
+	icon_state = src.icon_broken
+	user.visible_message("<span class='danger'>\The [src] has been broken by \the [user] with an electromagnetic card!</span>", "<span class='notice'>You break open \the [src].</span>", "<span class='notice'>You hear a faint click sound.</span>")
```

Comment:
uh

Reply:
Thanks, the previous implementation had a view of 3 and not of 7.